### DISCUSSION TITLE
Support for Windows static site generation - anyone have a current workaround?

### DISCUSSION THREAD

**Discussion by textplusdata** [datenstrom/community#149] [Ask a question] [answered]
Hi,

I read the issues connected to the problems with static generation on Windows server:
commandline build cannot write files? - Windows datenstrom/community#115
Static pages not show 'error.html' when sth missing. datenstrom/community#116
Better static build datenstrom/community#16
Unusual pagination arguments datenstrom/community#65
Enhance static website generator datenstrom/community#125 

What I was wondering is if anyone has a workaround for this at the moment? I tried re-writing the commandline.php plugin to give the error code but then run the build anyway. The build ran without any errors (on vanilla Yellow install, so no blog, just the "about" and "home" pages, but I couldn't find the created files anywhere at all on the system.

If I've understood the previous issues, the Build command for a static site should only fail on the blog plugin where the colon is used. Should I expect to be able to build a static site on Windows as long as I am not using the blog plugin?

Thanks!

**Comment by markseuffert** [comment]
Hi, what happens if you change the following in `system/plugins/commandline.php` file and then run the build command `php yellow.php build test`. Can you find the generated files, there should be a new folder in your Yellow folder? Can you copy/paste the command line output? 

```
function checkStaticFilesystem()
{
    return true; //just a workaround
}
```

To answer your other question. Yes you're right, you should be able to [generate a static website](https://github.com/annaesvensson/yellow-generate) on Windows with this workaround, for example a default Yellow installation. The blog and other plugins will fail with an error 500 "Can't write file", if they make use of the colon character.

**Comment by markseuffert** [comment]
If you're a developer and like to see more information what's going on, you can [enable debug mode](https://datenstrom.se/yellow/help/troubleshooting#problems-after-installation).

Hope this helps.

**Comment by textplusdata** [comment, marked as answer]
Hi,

Thanks for replying. I tested again and now realise I've been a bit slow.

This is the result

```
Yellow build: 3 content, 2 media, 4 system, 0 errors, status 200
```

My config.ini file has this (I am running on a local server on my windows box)

```
ServerScheme: http
ServerName: localhost
ServerBase: /yellow-master
ServerTime: Europe/London
```

My install is here:
D:\UniServerZ\www\yellow-master

so I was expecting to see the results of 

```
php yellow.php build /var/www/website
```

in
D:\UniServerZ\www\yellow-master/var/www/website

Instead they are here at the root of the drive
D:\var\www\website

For anyone else who might make this mistake, you would need to miss off the backslash off the folder name if you want the static files in the same folder (which you probably wouldn't). This is actually already indicated in the documentation with an example of exporting to the cache folder where there is no backslash before the folder name "cache"
https://github.com/datenstrom/yellow/wiki/Web-server-configuration#static-website

```
php yellow.php build cache /blog/blog-example
```

Should I open a new issue to see if I can get tips from people in terms of changing the separator from ":" to "--" in the blog file?  If so, you can close this, thanks!

I was thinking that it make sense to roll this kind of thing into the config where you can specify a separator and the default is ":", which would allow windows people to change it to "--" or whatever they want? I have rudimentary php so with a bit of a pointer about setting up some kind of regex in the code somewhere I guess I could at least have a go :-)

**Comment by markseuffert** [comment]
Good to hear it works for you. :smile: 

About changing the separator, @wunderfeyd and myself came to the same conclusion. It could be a configuration setting that core and plugins use, for example something like `:` or `=`. We shouldn't expect that Microsoft makes their file system more web developer friendly, so a configuration setting might solve the problem for Windows users. Please open a new issue, if you want to suggest this feature. 

Have a look around in the source code, any help improving Yellow would be much appreciated.
